pubtools/_pulp/tasks/push/items/erratum.py

```diff
@@ -29,13 +29,17 @@ class ErratumUploadContext(UploadContext):
 class PulpErratumPushItem(PulpPushItem):
     """Handler for errata."""
 
-    # Erratums are always uploaded to this repo first.
-    # Pulp stores the erratum pkglist for (erratum_id, repo) pair. If an erratum is
-    # uploaded for an already existing ID, the pkglist is overwritten if the repo is
-    # same, else is appended. To keep the pkglist consistent, all the erratums are
-    # uploaded to the same repo instead of a random repo from the list of destinations.
-    # For more details, refer RHELDST-12782.
-    UPLOAD_REPO = "all-rpm-content"
+    MULTI_UPLOAD_CONTEXT = True
+
+    @property
+    def upload_repo(self):
+        # Split errata into different repos by year, assuming the advisory name
+        # is formatted like RHXA-YYYY
+        try:
+            return "all-erratum-content-%s" % self.pushsource_item.name[5:9]
```

Comment:
I think we should at least have some minimal validation of this, not necessarily to handle it, but at least to make it crash ASAP if our expectation is not met. Can we please do that?

For example, just wrapping it with `int(self.pushsource_item.name[5:9])` to crash if the string portion is not numeric would be good enough for me.  Then an unexpected advisory name would cause it to crash early and immediately draw attention to this code.

If it's not validated at all, an unexpected advisory name can cause it to try uploading to a nonexistent `all-erratum-content-something` repo.  A developer looking at an error like that is pretty likely to waste time thinking that the error is that the repo is expected to exist but doesn't, and it may take them a while to figure out that the real problem was that this code received an unexpected advisory name.

Reply:
Rohan, the more I think about it, the more I feel like pubtools-pulp should check the year ranges. It's an implementation detail specific to our version of pulp, but without some checking we wont end up using the default repo at all. Do we know how likely it is we'll be dealing with errata older than 2014?

Reply:
I think that's pretty unlikely.  I think 2016 was the birth of the current way of doing things, so when I mentioned 2014 as the earliest year I was already adding a couple of extra years to be safe.